app/src/main/java/com/android/shelfLife/viewmodel/overview/HouseholdCreationScreenViewModel.kt

```diff
@@ -30,127 +31,115 @@ constructor(
   val households = houseHoldRepository.households
   val currentUser = userRepository.user
 
-  var finishedLoading = MutableStateFlow(false)
+  private var finishedLoading = MutableStateFlow(false)
 
-  // Fetch the list of members for the household to edit and set the email list
   init {
     viewModelScope.launch {
       val members = houseHoldRepository.getHouseholdMembers(householdToEdit.value?.uid ?: "")
-      userRepository.getUserEmails(members) { uidToEmail ->
-        _emailList.value = uidToEmail.values.toSet()
-      }
+      _emailList.value = userRepository.getUserEmails(members).values.toSet()
       finishedLoading.value = true
     }
   }
 
-  /**
-   * Sets the list of emails to the provided list. FOR TESTING PURPOSES ONLY.
-   *
-   * @param emails - The list of emails to set.
-   */
-  fun setEmails(emails: Set<String>) {
-    _emailList.value = emails
-  }
-
-  fun addEmail(email: String) {
+  private fun addEmail(email: String) {
     _emailList.value += email
   }
 
   fun removeEmail(email: String) {
     _emailList.value -= email
   }
 
-  /**
-   * Gets the user IDs corresponding to a list of emails.
-   *
-   * @param emails - The list of emails to get user IDs for.
-   * @param callback - The callback to be invoked with the map of email to user ID.
-   */
-  fun getUserIdsByEmails(emails: Set<String>, callback: (Map<String, String>) -> Unit) {
-    userRepository.getUserIds(emails) { emailToUid -> callback(emailToUid) }
+  private fun newHouseholdNameIsInvalid(householdName: String): Boolean {
+    return (householdName.isBlank() ||
+            (houseHoldRepository.checkIfHouseholdNameExists(householdName) &&
+                    (householdToEdit.value == null || householdName != householdToEdit.value!!.name)))
   }
 
-  fun newHouseholdNameIsInvalid(householdName: String): Boolean {
-    return (householdName.isBlank() ||
-        houseHoldRepository.checkIfHouseholdNameExists(householdName) &&
-            (householdToEdit.value == null || householdName != householdToEdit.value!!.name))
+  private suspend fun getEmailToUserId(emails: Set<String>): Map<String, String> {
+    return userRepository.getUserIds(emails)
   }
 
   /**
-   * Adds a new household to the repository and updates the household list.
-   *
-   * @param householdName - The name of the household to be added.
+   * Attempts to add an email card, returning true if successful and false otherwise.
+   * This encapsulates the logic for checking duplicates and blank inputs.
    */
-  fun addNewHousehold(householdName: String, friendEmails: Set<String?> = emptySet()) {
-    if (currentUser.value != null) {
+  fun tryAddEmailCard(emailInput: String): Boolean {
+    val trimmedEmail = emailInput.trim()
```

Comment:
I see that we are now prefix the emails, is this more efficient than just he while emailInput?